core/os/android/adb/commands.go

```diff
@@ -349,10 +349,13 @@ func (b *binding) QueryPerfettoServiceState(ctx context.Context) (*device.Perfet
 	}
 	result.GpuProfiling = gpu
 
-	// SurfaceFlinger frame lifecycle perfetto producer is mandated by Android 11 CTS, hence it will
-	// always exist.
 	if b.Instance().GetConfiguration().GetOS().GetAPIVersion() >= 30 {
+		// SurfaceFlinger frame lifecycle perfetto producer is mandated by Android 11 CTS, hence it will
+		// always exist.
 		result.HasFrameLifecycle = true
+
+		// This has anecdotally not worked well in Q, but appears to be fine in R.
```

Comment:
I think we should prefer `Android11` to Q and `Android12` to R in general, and here the comment above already uses "Android 11". I would be keen on having "Android" and its version number in one word, to help grepping for it in the future.

Reply:
You pry the letter names from my cold dead fingers.